seb_openedx/middleware.py

```diff
@@ -1,27 +1,132 @@
+
+        if 'discussion' in SEB_WHITELIST_PATHS and views_module == 'discussion.views':
+            return True
+
+        if 'wiki' in SEB_WHITELIST_PATHS and views_module == 'course_wiki.views':
+            return True
+
+        if 'courseware' in SEB_WHITELIST_PATHS and self.is_xblock_request(request):
+            return True
+
+        url_names_allowed = list(SEB_WHITELIST_PATHS)
+
+        if 'course-outline' in SEB_WHITELIST_PATHS:
+            url_names_allowed.append('openedx.course_experience')
+
+        if request.resolver_match.url_name:
+            if request.resolver_match.url_name in ['jump_to', 'jump_to_id']:
+                # Allow redirects:
+                return True
+            for url_name in url_names_allowed:
+                if request.resolver_match.url_name.startswith(url_name):
+                    return True
+        return False
+
+    def is_blacklisted_chapter(self, request, course_key):
+        """ Second more granular filter: blacklisting of specific chapters """
+        chapter = request.resolver_match.kwargs.get('chapter')
+
+        if not SEB_BLACKLIST_CHAPTERS:
+            return False
+
+        if chapter in SEB_BLACKLIST_CHAPTERS:
+            return True
+
+        if 'courseware' in SEB_WHITELIST_PATHS and self.is_xblock_request(request):
+            modulestore = get_xmodule_modulestore_module().django.modulestore
+            path_to_location = get_xmodule_modulestore_module().search.path_to_location
+            usage_id = request.resolver_match.kwargs.get('usage_id')
+            usage_key = UsageKey.from_string(unquote_slashes(usage_id)).map_into_course(course_key)
+            if usage_key:
+                path = path_to_location(modulestore(), usage_key)
+                chapter = path[1]
+                if chapter in SEB_BLACKLIST_CHAPTERS:
+                    return True
+        return False
+
+    def courseware_error_response(self, request, *view_args, **view_kwargs):
+        """ error response when a chapter is being blocked """
+        html = Fragment()
+        html.add_content(render_to_string('seb-403-error-message.html', {}))
+        SebCoursewareIndex.set_context_fragment(html)
+        return SebCoursewareIndex.as_view()(request, *view_args, **view_kwargs)
+
+    def generic_error_response(self, request, course_key):
+        """ generic error response, full page 403 error (with course menu) """
+        courseware = get_courseware_module()
+        try:
+            course = courseware.courses.get_course(course_key, depth=2)
+        except ValueError:
+            return HttpResponseNotFound()
+
+        context = {
+            'course': course,
+            'request': request
+        }
+
+        return render_to_response('seb-403.html', context, status=403)
+
+    def is_xblock_request(self, request):
+        """ returns if it's an xblock HTTP request or not """
+        return request.resolver_match.func.__name__ == 'handle_xblock_callback'
```

Comment:
This returns true only for requests of the form: `/courses/<ID>/xblock/block-v1:<ID>+type@type_of_xblock+block@<SOMEHASH>/handler/name_of_handler` or are we catching something else?

Reply:
I think that's all we need to match for xblocks